### PR TITLE
Create automated release process

### PR DESCRIPTION
**Pull request by normartin**
Provide an automated release process using Github actions

- publishes to npmjs.com
- creates Git tag

Improves build process by avoiding duplicated CI runs on pull requests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -209,3 +209,9 @@ it("can customize default config", async () => {
     }
 });
 ```
+
+# Release instructions
+Release automation has been setup according this [guide](https://michaelzanggl.com/articles/github-actions-cd-setup/).
+
+1. Create a Github release with version tag like `0.6.1`. 
+1. Check the new version exists on [npmjs.com/package/ts-retry-promise](https://www.npmjs.com/package/ts-retry-promise) and has `latest` tag.
```
Comment (memark) [inline README.md:217]: This should probably be 1. and 2.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      - uses: actions/checkout@v2
+        with:
+          # "ref" specifies the branch to check out.
+          # "github.event.release.target_commitish" is a global variable and specifies the branch the release targeted
+          ref: ${{ github.event.release.target_commitish }}
+      # install Node.js
+      - name: Use Node.js 12
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
```
Comment (memark) [inline .github/workflows/release.yml]: Consider using v14, as it's the current LTS.

**Review by memark — APPROVED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (normartin) [reply README.md:217]: I tend to use only 1.s as it makes inserting and reordering simpler. Will be render as 1.,2.,...

**Comment by normartin** [comment]
@memark thanks for looking into this!

File: README.md (hunk 1 above)
Comment (memark) [reply README.md:217]: Ah, didn't know that worked!

**Pull request merged**